test/formal/and_distributed_over_shl.py

```diff
@@ -0,0 +1,25 @@
+from opcodes import AND, SHL
+from rule import Rule
+from z3 import BitVec
+
+"""
+Rule:
+AND(SHL(A,X), SHL(A,Y)) -> SHL(A, AND(X,Y))
+"""
+
+rule = Rule()
+
+n_bits = 128
```

Comment:
why not 256?

Reply:
We're doing a Shift Left Operation being performed below. Having a 128 helps avoid overflow and keeps it simple. 

But if I've missed something and its required, I can change it to 256 bits as well.

Reply:
Right. I was just curious about it.

Reply:
If it works using full 256 bits, i.e. it actually proves it, then it should *always* be 256 bits.
Even more so since using a lower bit width exactly might hide overflow cases or special edge cases. Usually the proven properties are monotone in bit-width and at times the solvers can't prove them at full 256-bits, but it significantly increases the review effort not to use 256 bits, so: if possible always keep at 256 bits.

Reply:
Just as a general remark. I just checked it here: the rule works for 256 bits, but is very slow to prove - and it's rather easy to see that it's monotonic in bit width, so it's fine to keep it at 128.